// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.



syntax = "proto3";

package grpc;

message Empty {}

// I called it Note because message Message sounds complicated
message Note {
    string name = 1;
    string message = 2;
    string group=3;
}
message GroupNote {
    string name = 1;
    string message = 2;
    string group=3;
}

message registerdetails{
    string username=1;
    string password=2;
    string email=3;
}
message LoginRequest{

    string username=1;
    string password=2;

}
message APIResponse{
     string responsemessage=1;
     int32 responsecode=2;
}

message ChatUserMessage{
    string name = 1;
    string message = 2;
    string receiver=3;
}
message key{
    string name=1;
}

service ChatServer {
    rpc ChatStream (Empty) returns (stream Note);
    rpc SendNote (Note) returns (Empty);
    rpc SendNotemulti (ChatUserMessage) returns (Empty);
    rpc register(registerdetails) returns(APIResponse);
    rpc login(LoginRequest) returns(APIResponse);
    rpc logout(Empty) returns (APIResponse);
    rpc ChatUser(ChatUserMessage) returns(stream Note);
    rpc ChatStreamUser (Empty) returns (stream ChatUserMessage);
    rpc RegisteredUsers(Empty) returns(stream registerdetails);
}